### DISCUSSION TITLE
DAG scheduler

### DISCUSSION THREAD

**Discussion by Ausutkarsh** [apache/airflow#31126] [Q&A]
there is dag having 5 tasks 
scheduled for hourly 
rough estimate to finish dag is 17 minutes,
task 1 successfully completed
task 2 triggers 2nd dag but dag 1 does not wait for 2nd dag to be completed and moves to task 3
task 3 successfully completed
task 4 triggers 3rd  dag but dag 1 does not wait for 3rd  dag to be completed and moves to task 5
rough estimate for  2nd dag to be completed is 10 minutes
rough estimate for  3rd  dag to be completed is 10 minutes
but sometimes 2nd dag or 3rd dag takes more than estimated time and sum of these 3 dags to be completed gets more than 1 hour
so what happens as time completes 1 hour dag 1 start running 
but this should not happen
till all 3 dags are completed dag 1 should not run even it’s it scheduled, it should skip it’s run.
DAG1--> raw layer
DAG2--> staging layer
DAG3--> processing layer

there is no direct dependency of dag2 and dag3.
**Question **--> 2nd run of DAG 1 should only initiate when 1st run of DAG2 and DAG 3 are complete else 2nd run of dag1 must be skipped no matter it is scheduled.

**Comment by potiuk** [comment]
Is there a question in this stream of text?

**Reply by Ausutkarsh** [reply]
@potiuk, I have updated the issue. Thanks in advance

**Reply by potiuk** [reply]
I can't understand the question/problem.
